spinalcordtoolbox/deepseg/models.py

```diff
@@ -443,11 +461,16 @@ def install_model(name_model, custom_url=None):
     else:
         if not isinstance(url_field, dict):
             raise ValueError("Invalid url field in MODELS")
-        urls_used = {}
-        for seed_name, model_urls in url_field.items():
-            logger.info(f"\nInstalling '{seed_name}'...")
-            urls_used[seed_name] = download.install_data(model_urls,
-                                                         folder(os.path.join(name_model, seed_name)), keep=True)
+        # totalspineseg handles data downloading itself, so just pass the urls along
+        if name_model == 'totalspineseg':
+            tss_init.init_inference(data_path=Path(folder(name_model)), quiet=False, dict_urls=url_field)
```

Comment:
It looks like the tests are failing on Windows due to a FileNotFound error for this call:

```
    subprocess.run(['nnUNetv2_install_pretrained_model_from_zip', str(zip_file)])
```

I think maybe this needs to be `nnUNetv2_install_pretrained_model_from_zip.exe` on Windows? 

To avoid cross-compatibility issues, I would double-check to see if there is a way to run the `nnunetv2` commands via the Python API, rather than using `subprocess`. Otherwise, you could maybe just check if the OS is Windows, then append `.exe` to all of your subprocess calls.

Reply:
I added the `.exe` as requested on the subprocess calls

Reply:
Huh! I'm surprised... the Windows tests are still failing: https://github.com/spinalcordtoolbox/spinalcordtoolbox/actions/runs/11959633540/job/33341854040?pr=4690#step:6:916

Maybe my hunch was wrong, then? I'll try to debug on a Windows machine, maybe...

Reply:
I mean, the script is listed:

![image](https://github.com/user-attachments/assets/55bd8f52-80fe-4dbf-99ac-973ef029e1d4)

But it's not on the PATH... hmm... Oh!! I know what's happening here.

- On Linux/macOS, the folder with the CLI scripts is called `bin/`
- On Windows, the folder is called `Scripts/`

When I check `os.environ['PATH']` on Linux, I see:

```
/home/joshua/repos/spinalcordtoolbox/python/envs/venv_sct/bin
```

However, when I check `os.environ['PATH']` on Windows, I see:

```console
PS C:\Users\joshu\sct_6.4> python/envs/venv_sct/python.exe -c "import os; print(os.environ['PATH'])"
C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\Program Files\Git\cmd;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\bin;C:\Users\joshu\AppData\Local\Microsoft\WindowsApps;C:\Users\joshu\sct_6.4\bin\;
```

So, for some reason, on Linux, the path containing the nnUNet executable is on the PATH. But, on Windows, the corresponding directory is _not_ on the path. I'm not sure why, but that certainly explains it.

The reason that TotalSpineSeg has worked in the past is because the README.md asks users to activate their venv. But, for SCT, we don't have that luxury. So, right now `totalspineseg` is broken as a PyPI package due to its use of subprocess...

Reply:
I would recommend swapping out the subprocess calls for calls to Python API calls, [for example](https://github.com/MIC-DKFZ/nnUNet/blob/ac79a612ae696af224169cbefde7519354782e64/nnunetv2/model_sharing/model_import.py#L6):

```python
from nnunetv2.model_sharing import install_model_from_zip_file
```

This will work for all platforms, and will save us from having to worry about the PATH.

Reply:
I just talked with @jcohenadad and maybe we should consider not being compatible with windows users yet as more problems may occurs and because it may take too much time solving all of them before the release. The original goal was to have a temporary implementation before having a complete integration inside SCT. So maybe we can raise a warning for windows users.

Reply:
Unfortunately it seems like the "FileNotFound" error is not Windows-specific... I think this will actually block the PR for Linux, too, unless it is fixed: https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4690#discussion_r1854013930

Reply:
Maybe we can add a quick-fix to add the `bin/` (UNIX) and `Scripts/` (Windows) folders to the PATH?

I don't like this solution, but it will help us avoid having to rewrite the TotalSpineSeg package. I'll submit the fix now to test it out.